assets/js/lib/test-utils/factories/executions.js

```diff
@@ -83,3 +113,70 @@ export const withCompletedResults = (execution, result, checkResults) => ({
 export const checksExecutionCompletedFactory = Factory.define(({ params }) =>
   withCompletedResults(checksExecutionRunningFactory.build(params))
 );
+
+export const withEmptyExpectations = (checkResult) => {
+  const agents = checkResult.agents_check_results.map((agent) => ({
+    ...agent,
+    expectation_evaluations: [],
+  }));
+
+  return {
+    ...checkResult,
+    agents_check_results: agents,
+    expectation_results: [],
+  };
+};
+
+const addExpectation = (checkResult, name, expec, result) => {
+  const { type } = expec;
+  const agents = checkResult.agents_check_results.map((agent) => {
+    const evals = [...agent.expectation_evaluations, expec];
+
+    return {
+      ...agent,
+      expectation_evaluations: evals,
+    };
+  });
+
+  const results = [
+    ...checkResult.expectation_results,
+    expectationResultFactory.build({ name, result, type }),
+  ];
+
+  return {
+    ...checkResult,
+    agents_check_results: agents,
+    expectation_results: results,
+  };
+};
+
+export const addPassingExpectation = (checkResult, type) => {
+  const name = faker.company.name();
+  const expec = executionExpectationEvaluationFactory.build({
+    name,
+    type,
+    return_value: true,
+  });
+
+  return addExpectation(checkResult, name, expec, true);
+};
+
+export const addCriticalExpectation = (checkResult, type) => {
+  const name = faker.company.name();
+  const expec = executionExpectationEvaluationFactory.build({
+    name,
+    type,
+    return_value: false,
+  });
+
+  return addExpectation(checkResult, name, expec, false);
+};
+
+export const addExpectationWithError = (checkResult) => {
```

Comment:
maybe `addErroredExpectation` sound better and matches the same wording as the other functions?